plugins/graylog/graylog_input.go

```diff
@@ -0,0 +1,120 @@
+package graylog
+
+import (
+	"strings"
+
+	"github.com/pborman/uuid"
+
+	"github.com/mozilla-services/heka/pipeline"
+	"github.com/mozilla-services/heka/message"
+
+	"github.com/Graylog2/go-gelf/gelf"
+)
+
+type GraylogInputConfig struct{
+	Address string `toml:"address"`
+}
+
+type GraylogInput struct {
+	config *GraylogInputConfig
+	reader *gelf.Reader
+
+	ctrlMsgs chan gelfCtrl
+	stopChan chan bool
+}
+
+func (g *GraylogInput) ConfigStruct() interface{} {
+	return &GraylogInputConfig{
+	}
+}
+
+func (g *GraylogInput) Init(config interface{}) (err error) {
+	g.config = config.(*GraylogInputConfig)	
+	g.ctrlMsgs = make(chan gelfCtrl)
+	g.stopChan = make(chan bool)
+	g.reader,err = gelf.NewReader(g.config.Address)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+type gelfCtrl struct {
+	err error
+	message *gelf.Message
+}
+
+func (g *GraylogInput) Run(ir pipeline.InputRunner, h pipeline.PluginHelper) (err error) {
+	go func() {
+		for {
+			select {
+			case <-g.stopChan:
+				break
```

Comment:
This break statement (and the one below on L61) will only break from the select, which means that AFAICT the for loop will never exit.